lib/restful_resource/strict_open_struct.rb

```diff
@@ -0,0 +1,5 @@
+module RestfulResource
+  class StrictOpenStruct < ::OpenStruct
+    undef_method :dig, :[]
```

Comment:
I would imagine that we use `dig` and `[]` in many places so maybe doing it as a deprecation warning as you said to start with would be less annoying? And then we can create some TI cards for that.